manifests/resource/location.pp

```diff
@@ -153,66 +153,65 @@
+  String $proxy_read_timeout                          = $::nginx::proxy_read_timeout,
+  String $proxy_connect_timeout                       = $::nginx::proxy_connect_timeout,
+  Array $proxy_set_header                             = $::nginx::proxy_set_header,
+  Array $proxy_hide_header                            = $::nginx::proxy_hide_header,
+  Array $proxy_pass_header                            = $::nginx::proxy_pass_header,
+  Optional[String] $fastcgi                           = undef,
+  Optional[String] $fastcgi_index                     = undef,
+  Optional[Hash] $fastcgi_param                       = undef,
+  String $fastcgi_params                              = "${::nginx::conf_dir}/fastcgi_params",
+  Optional[String] $fastcgi_script                    = undef,
+  Optional[String] $fastcgi_split_path                = undef,
+  Optional[String] $uwsgi                             = undef,
+  Optional[Hash] $uwsgi_param                         = undef,
+  String $uwsgi_params                                = "${nginx::config::conf_dir}/uwsgi_params",
+  Optional[String] $uwsgi_read_timeout                = undef,
+  Boolean $ssl                                        = false,
+  Boolean $ssl_only                                   = false,
+  Optional[String] $location_alias                    = undef,
+  Optional[Enum['any', 'all']] $location_satisfy      = undef,
+  Optional[Array] $location_allow                     = undef,
+  Optional[Array] $location_deny                      = undef,
+  Optional[Boolean ] $stub_status                     = undef,
+  Optional[Variant[String, Array]] $raw_prepend       = undef,
+  Optional[Variant[String, Array]] $raw_append        = undef,
+  Optional[Hash] $location_custom_cfg                 = undef,
+  Optional[Hash] $location_cfg_prepend                = undef,
+  Optional[Hash] $location_cfg_append                 = undef,
+  Optional[Hash] $location_custom_cfg_prepend         = undef,
+  Optional[Hash] $location_custom_cfg_append          = undef,
+  Optional[Array] $include                            = undef,
+  Optional[Array] $try_files                          = undef,
+  Optional[String] $proxy_cache                       = undef,
+  Optional[String] $proxy_cache_key                   = undef,
+  Optional[String] $proxy_cache_use_stale             = undef,
+  Optional[Variant[Array, String]] $proxy_cache_valid = undef,
+  Optional[String] $proxy_method                      = undef,
+  Optional[String] $proxy_http_version                = undef,
+  Optional[String] $proxy_set_body                    = undef,
+  Optional[Enum['on', 'off']] $proxy_buffering        = undef,
+  Optional[String] $auth_basic                        = undef,
+  Optional[String] $auth_basic_user_file              = undef,
+  Optional[String] $auth_request                      = undef,
+  Array $rewrite_rules                                = [],
```

Comment:
it was already an existing array. I didn't want to change the datatype to undef and make it optional.